cmd/podman/machine/ssh.go

```diff
@@ -44,6 +47,14 @@ func ssh(cmd *cobra.Command, args []string) error {
 
 	// Set the VM to default
 	vmName := defaultMachineName
+
+	// If we're not given a VM name, use the remote username from the connection config
+	if len(args) == 0 {
+		sshOpts.Username, err = remoteConnectionUsername()
+		if err != nil {
+			return err
+		}
```

Comment:
Why is this set in three places? It should be enough to set this in one place.

Reply:
Each of these checks a different condition. 
`podman machine ssh` should be set
`podman machine ssh vmname` should not be set vs `podman machine ssh sleep` should be set
`podman machine ssh vmname sleep ` should not be set vs `podman machine ssh echo "hello"` should be set
```